processing/src/main/java/io/druid/segment/StringDimensionHandler.java

```diff
@@ -0,0 +1,1248 @@
+{
+  private static final Logger log = new Logger(StringDimensionHandler.class);
+
+  private static final ListIndexed EMPTY_STR_DIM_VAL = new ListIndexed<>(Arrays.asList(""), String.class);
+  protected static final Splitter SPLITTER = Splitter.on(",");
+
+  private final String dimensionName;
+
+  public StringDimensionHandler(String dimensionName)
+  {
+    this.dimensionName = dimensionName;
+  }
+
+  @Override
+  public DimensionSelector getDimensionSelector(ColumnSelectorFactory cursor, DimensionSpec dimSpec, ColumnCapabilities capabilities)
+  {
+    return cursor.makeDictEncodedStringDimensionSelector(dimSpec);
+  }
+
+  @Override
+  public Function<Object, String> getValueTypeTransformer()
+  {
+    return STRING_TRANSFORMER;
+  }
+
+  @Override
+  public Comparator<Integer> getEncodedComparator()
+  {
+    return ENCODED_COMPARATOR;
+  }
+
+  @Override
+  public Comparator<String> getActualComparator()
+  {
+    return UNENCODED_COMPARATOR;
+  }
+
+  @Override
+  public byte[] getBytesFromRowValue(Integer rowVal)
+  {
+    return Ints.toByteArray(rowVal);
+  }
+
+  @Override
+  public Integer getRowValueFromBytes(byte[] bytes)
+  {
+    return Ints.fromByteArray(bytes);
+  }
+
+  @Override
+  public int getEncodedValueSize()
+  {
+    return Ints.BYTES;
+  }
+
+  @Override
+  public DimensionIndexer makeIndexer(Object lock)
+  {
+    return new StringDimensionIndexer(lock);
+  }
+
+  @Override
+  public DimensionMerger makeMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      FileSmoosher v9Smoosher,
+      ObjectMapper mapper,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  )
+  {
+    return new StringDimensionMerger(indexSpec, outDir, ioPeon, v9Smoosher, mapper, capabilities, progress);
+  }
+
+  @Override
+  public DimensionColumnReader makeColumnReader(Column column)
+  {
+    return new StringDimensionColumnReader(column);
+  }
+
+  public static final Function<Object, String> STRING_TRANSFORMER = new Function<Object, String>()
+  {
+    @Override
+    public String apply(final Object o)
+    {
+      if (o == null) {
+        return null;
+      }
+      if (o instanceof String) {
+        return (String) o;
+      }
+      return String.valueOf(o);
+    }
+  };
+
+  private static final Comparator<Integer> ENCODED_COMPARATOR = new Comparator<Integer>()
+  {
+    @Override
+    public int compare(Integer o1, Integer o2)
+    {
+      if (o1 == null) {
+        return o2 == null ? 0 : -1;
+      }
+      if (o2 == null) {
+        return 1;
+      }
+      return Integer.compare(o1, o2);
+    }
+  };
+
+  private static final Comparator<String> UNENCODED_COMPARATOR = new Comparator<String>()
+  {
+    @Override
+    public int compare(String o1, String o2)
+    {
+      if (o1 == null) {
+        return o2 == null ? 0 : -1;
+      }
+      if (o2 == null) {
+        return 1;
+      }
+      return ObjectUtils.compare(o1, o2);
```

Comment:
ObjectUtils.compare(o1,o2) handles null as well so the null checks before are redundant or replace this with just
o1.compareTo(o2)